### PR TITLE
Nesterov Momentum

### PR DESCRIPTION
**Pull request by Thrandis**
Fixes mila-udem/blocks#762

### CODE REVIEW

**Comment by Thrandis** [comment]
@dmitriy-serdyuk, there you go!

File: blocks/extras/algorithms/__init__.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    -----
+    This step rule is intended to be used in conjunction with another
+    step rule, _e.g._ :class:`Scale`. For an all-batteries-included
+    experience, look at :class:`NesterovMomentum`.
+
+    It is implemented as in [BBR13].
+
+    .. [BBR13] Yoshua Bengio, Nicolas Boulanger-Lewandowski, and Razvan
+        Pascanu, *Advances in optimizing recurrent networks*,
+        ICASSP (2013), pp 8624-8628.
```
Comment (dmitriy-serdyuk) [inline blocks/extras/algorithms/__init__.py]: Can you also add links to original papers? Seems, that it was introduced in Nesterov, Y.  A method of solving a convex programming problem with convergence rate O (1/k2)

**Comment by Thrandis** [comment]
@dmitriy-serdyuk :-)

File: tests/test_algorithms.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from blocks.utils import shared_floatx
+
+
+def test_basic_nesterov_momentum():
+    a = shared_floatx([3, 4])
+    cost = (a ** 2).sum()
+    steps, updates = BasicNesterovMomentum(0.5).compute_steps(
+        OrderedDict([(a, tensor.grad(cost, a))]))
+    f = theano.function([], [steps[a]], updates=updates)
+    assert_allclose(f()[0], [9., 12.])
```
Comment (dmitriy-serdyuk) [inline tests/test_algorithms.py]: Can you compile another function, which computes the update like in equations (1)-(2) of the paper and compare here that simplified NAG returns the same result?

**Comment by Thrandis** [comment]
@dmitriy-serdyuk is it ok like this?

**Comment by dmitriy-serdyuk** [comment]
Looks good!

**Pull request merged**